lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2419,7 +2410,7 @@ static void checkAccessControl(TypeChecker &TC, const Decl *D) {
     if (!EED->hasAssociatedValues())
       return;
     for (auto &P : *EED->getParameterList()) {
-      checkTypeAccess(TC, P->getTypeLoc(), P,
+      checkTypeAccess(TC, P->getTypeLoc(), EED->getParentEnum(),
```

Comment:
This one seems wrong; it should just be `EED`, right? (Because some day we might have enum cases less visible than the enum.)